app/feature/feature-addon-purchase/src/main/kotlin/com/hedvig/android/feature/addon/purchase/ui/summary/AddonSummaryDestination.kt

```diff
@@ -0,0 +1,287 @@
+        confirmButtonLabel = stringResource(R.string.ADDON_FLOW_CONFIRMATION_BUTTON),
+        dismissButtonLabel = stringResource(R.string.general_close_button),
+        text = stringResource(
+          R.string.ADDON_FLOW_CONFIRMATION_DESCRIPTION,
+          formattedDate,
+        ),
+      )
+    }
+    SummaryCard(
+      uiState = uiState,
+      modifier = Modifier
+        .fillMaxWidth()
+        .padding(16.dp),
+    )
+    Spacer(Modifier.weight(1f))
+    Spacer(Modifier.height(16.dp))
+    Column(
+      Modifier
+        .padding(horizontal = 16.dp),
+    ) {
+      HedvigNotificationCard(
+        message = stringResource(R.string.ADDON_FLOW_SUMMARY_INFO_TEXT),
+        priority = NotificationDefaults.NotificationPriority.Info,
+      )
+      Spacer(Modifier.height(40.dp))
+      HorizontalItemsWithMaximumSpaceTaken(
+        startSlot = {
+          HedvigText(
+            stringResource(R.string.TIER_FLOW_TOTAL), // todo: change copy prob?
+            style = HedvigTheme.typography.bodySmall,
+          )
+        },
+        spaceBetween = 8.dp,
+        endSlot = {
+          HedvigText(
+            text = stringResource(
+              R.string.ADDON_FLOW_PRICE_LABEL,
+              uiState.summaryParameters.quote.price.amount.toInt(),
+            ),
+            textAlign = TextAlign.End,
+            style = HedvigTheme.typography.bodySmall,
+          )
+        },
+      )
+      Row(
+        Modifier.fillMaxWidth(),
+        horizontalArrangement = Arrangement.End,
+      ) {
+        HedvigText(
+          text = stringResource(R.string.ADDON_FLOW_SUMMARY_PRICE_SUBTITLE),
+          textAlign = TextAlign.End,
+          style = HedvigTheme.typography.label,
+          color = HedvigTheme.colorScheme.textSecondary,
+        )
+      }
+      Spacer(Modifier.height(16.dp))
+      HedvigButton(
+        text = stringResource(R.string.ADDON_FLOW_SUMMARY_CONFIRM_BUTTON),
+        modifier = Modifier.fillMaxWidth(),
+        buttonStyle = Primary,
+        buttonSize = Large,
+        enabled = true,
+        onClick = {
+          showConfirmationDialog = true
+        },
+      )
+      Spacer(Modifier.height(16.dp))
+    }
+  }
+}
+
+@Composable
+private fun SummaryCard(uiState: Content, modifier: Modifier = Modifier) {
+  val locale = getLocale()
+  val formattedDate = remember(uiState.summaryParameters.activationDate, locale) {
+    HedvigDateTimeFormatterDefaults.dateMonthAndYear(
+      locale,
+    ).format(uiState.summaryParameters.activationDate.toJavaLocalDate())
+  }
+  QuoteCard(
+    subtitle = stringResource(R.string.ADDON_FLOW_SUMMARY_ACTIVE_FROM, formattedDate),
+    premium = uiState.summaryParameters.quote.price.toString(),
+    displayItems = uiState.summaryParameters.quote.addonVariant.displayDetails.map {
+      QuoteDisplayItem(
+        it.first,
+        null,
+        it.second,
+      )
+    },
+    underTitleContent = {
+      // todo: check here!
```

Comment:
Perhaps just fine for it to be nothing here if design has nothing under the price